K2Bridge/KustoDAL/KustoResponseParser.cs

```diff
@@ -168,6 +168,13 @@ private void ReportQueryExecutionMetrics(KustoResponseDataSet kustoResponseDataS
                 var (key, aggregationType) = query.PrimaryAggregation;
                 var tableData = kustoResponse[KustoTableNames.Aggregation].TableData;
 
+                // Not all aggregations return a metadata table
+                DataTable metadataTableData = null;
```

Comment:
I suppose you can replace all this block by:
var metadataTableData = kustoResponse[KustoTableNames.Metadata]?.TableData;

Reply:
Fixed